src/core/payments/index.ts

```diff
@@ -257,36 +257,27 @@ export async function depositUSDFC(
   synapse: Synapse,
   amount: bigint
 ): Promise<{
-  approvalTx?: string
   depositTx: string
 }> {
-  const paymentsAddress = synapse.getPaymentsAddress()
+  const needsAllowanceUpdate = (await checkAllowances(synapse)).needsUpdate
+  const amountMoreThanCurrentAllowance =
+    (await synapse.payments.allowance(synapse.getPaymentsAddress(), TOKENS.USDFC)) < amount
 
-  // Step 1: Check current allowance
-  const currentAllowance = await synapse.payments.allowance(paymentsAddress, TOKENS.USDFC)
+  let tx = {} as ethers.TransactionResponse
 
-  let approvalTx: string | undefined
-
-  // Step 2: Approve if needed (skip if already approved)
-  if (currentAllowance < amount) {
-    const approveTx = await synapse.payments.approve(paymentsAddress, amount, TOKENS.USDFC)
-    await approveTx.wait()
-    approvalTx = approveTx.hash
-  }
-
-  // Step 3: Make the deposit
-  const depositTransaction = await synapse.payments.deposit(amount, TOKENS.USDFC)
-  await depositTransaction.wait()
-
-  const result: { approvalTx?: string; depositTx: string } = {
-    depositTx: depositTransaction.hash,
+  if (amountMoreThanCurrentAllowance || needsAllowanceUpdate) {
+    tx = await synapse.payments.depositWithPermitAndApproveOperator(
+      amount,
+      synapse.getPaymentsAddress(),
```

Comment:
this needs to be the WarmStorageService address -- in this instance we're telling the payments contract to approve WarmStorage as an _operator_ of rails up to the specified allowance, without this, WarmStorage just gets rejected when trying to manage rails for this user